### DISCUSSION TITLE
Inhibit screensaver on fullscreen ?

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#5230] [Q&A] [answered]
I run `xset s 300` from `.xinitrc` that sets screensaver which blanks screen after 5 minutes.
I wanna inhibit that when i3's fullscreen triggered (Mod+F). Any ideas ?

I've tried `xidlehook`, which has `--not-when-fullscreen` option, but that only considers application level full screens (mpv, browser, etc), but not i3's mod+F on the applications such as terminal, etc.

**Comment by orestisfl** [comment]
It looks like `xidlehook` detects fullscreen via `_NET_WM_STATE_FULLSCREEN` which i3 does set. Have you tried using it? It looks like it should work.

**Reply by ghost** [reply]
Ah, I've just wondered about that. Looks like I've tested it for lockscreen. Not the screensaver.
But how to test it with screensaver ? What command should I have to use in xidlehook to inhibit only fullscreen ? Caz it will also inhibit on non-full screen cases too, if I set `xset s off`.

**Reply by ghost** [reply]
Oh wait. I'm gonna remove xset from .xinit & try `xset dpms force off` under xidlehook & letyouknow if it works...

**Comment by orestisfl** [comment]
To answer your question more generically, you can write your own script that subscribes to [i3 events](https://i3wm.org/docs/ipc.html). When the current window is fullscreen, you inhibit the screensaver and when it's not you re-enable it.

**Comment by ghost** [comment, marked as answer]
Tq. It worked. Just used dpms standby (which manually puts screen off on execution). Dropped screensaver thing, because it's behaviour is unknown.
`xidlehook --not-when-fullscreen --timer 180 'xset dpms force standby' ''`
